src/views/CompanyPanelView/Components/CompaniesTable.jsx

```diff
@@ -43,70 +45,36 @@ const useStyles = makeStyles(theme => ({
 }))
 
 export function CompaniesPanelTable () {
-  const { companies, isLoading, getCompanyState } = useCompanyPanel()
+  const [offset, setOffset] = useState(0)
+  const [rowsPerPage, setRowsPerPage] = useState(10)
+  const { total, companies, isLoading, getCompanyPanel } = useCompanyPanel({ limit: rowsPerPage, offset })
   const [wantsChange, setChange] = useState(false)
   const [page, setPage] = useState(0)
-  const [selected, setSelected] = useState([])
-  const [rowsPerPage, setRowsPerPage] = useState(10)
+  const { companiesToChange, isCompanyChecked, handleChange, cleanCompaniesToChange } = useCompaniesToChange()
   const classes = useStyles()
 
-  useEffect(() => {
-    mapPublicCompanies()
-  }, [companies])
-
-  const mapPublicCompanies = () => setSelected(companies.filter(company => company.is_public).map(company => company.id))
-
-  const handleChange = (event, companyId) => {
-    const checked = event.target.checked
-    setSelected(prev => {
-      if (checked) {
-        return [...prev, companies.find(company => company.id === companyId)?.id]
-      } else {
-        return prev.filter(id => id !== companyId)
-      }
-    })
-  }
-
-  const getChangedCompanies = (companies, selected) => {
-    return {
-      companies: companies.filter(company => {
-        // eslint-disable-next-line camelcase
-        const { is_public, id } = company
-        // eslint-disable-next-line camelcase
-        if ((is_public && selected.includes(id)) || (!is_public && !selected.includes(id))) {
-          return false
-        }
-        return true
-        // eslint-disable-next-line camelcase
-      }).reduce((prev, { id, is_public }) => ({ ...prev, [id]: !is_public }), {})
-    }
-  }
-
   const onSave = async (_) => {
-    const companiesToChange = getChangedCompanies(companies, selected)
-    if (Object.keys(companiesToChange.companies).length > 0) {
-      await changeCompanyPublicly(companiesToChange)
+    if (Object.keys(companiesToChange).length > 0) {
+      await changeCompanyPublicly({ companies: companiesToChange })
     }
     setChange(false)
-    await getCompanyState()
-  }
-
-  const isSelected = (company, selectedIds) => {
-    return selectedIds.some(id => id === company?.id)
+    await getCompanyPanel({ limit: rowsPerPage, offset })
   }
 
   const handleChangePage = (_event, newPage) => {
+    const nextOffset = newPage * rowsPerPage
```

Comment:
Same here.
Only for handle change page.
When user chooses more rows per page it's ok call the endpoint